src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_params.py

```diff
@@ -62,6 +62,13 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('no_push', help="Indicates whether the image built should be pushed to the registry.", arg_type=get_three_state_flag())
         c.argument('no_logs', help="Do not show logs after successfully queuing the build.", action='store_true')
 
+    with self.argument_context('acr image') as c:
+        c.argument('resource_id', help='The ARM resource ID of the source container registry.')
```

Comment:
Is the container registry referred to here always another ACR resource?  If so, this could just be called 'registry' or 'source_registry'.  It should also support names or IDs.  https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters

Reply:
It still needs to support the name-or-id paradigm of the CLI.  If it's a generic resource it's a bit more annoying than a strongly-typed resource like another ACR.

Reply:
It needs to enable our client side CLI to lookup resources cross subscription.

Reply:
To summarize the discussion we had with @djyou 
The goal is to make --source a required parameter which can be a partial image name and have -r as a parameter which will be used to derive the source which can be a name or dns or fully fully qualified resource id.